File: src/main/java/com/likelionknu/applyserver/recruit/data/repository/RecruitContentRepository.java
package com.likelionknu.applyserver.recruit.data.repository;

import com.likelionknu.applyserver.recruit.data.entity.RecruitContent;
import org.springframework.data.repository.query.Param;
import org.springframework.data.jpa.repository.Query;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;

import java.util.List;

public interface RecruitContentRepository extends JpaRepository<RecruitContent, Long> {

    List<RecruitContent> findByRecruitIdOrderByPriorityAsc(Long recruitId);
    List<RecruitContent> findAllByRecruit_IdAndRequiredTrue(Long recruitId);

    @Modifying
    @Query("delete from RecruitContent rc where rc.recruit.id = :recruitId")
    void deleteAllByRecruitId(@Param("recruitId") Long recruitId);
}

File: src/main/kotlin/com/likelionknu/applyserver/recruit/data/repository/RecruitContentRepository.kt
package com.likelionknu.applyserver.recruit.data.repository

import com.likelionknu.applyserver.recruit.data.entity.RecruitContent
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface RecruitContentRepository : JpaRepository<RecruitContent, Long> {

    fun findByRecruitIdOrderByPriorityAsc(recruitId: Long): List<RecruitContent>

    fun findAllByRecruit_IdAndRequiredTrue(recruitId: Long?): List<RecruitContent>

    @Modifying
    @Query("delete from RecruitContent rc where rc.recruit.id = :recruitId")
    fun deleteAllByRecruitId(@Param("recruitId") recruitId: Long)
}